src/rpc/misc.cpp

```diff
@@ -336,6 +336,89 @@ UniValue createmultisig(const JSONRPCRequest& request)
     return result;
 }
 
+
+UniValue createhtlc(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 4) {
+        string msg = "createhtlc seller_key refund_key hash timeout_type timeout\n"
+            "\nCreates an address whose funds can be unlocked with a preimage or as a refund\n"
+            "It returns a json object with the address and redeemScript.\n"
+
+            "\nArguments:\n"
+            "1. seller_key    (string, required) The public key of the possessor of the preimage.\n"
+            "2. refund_key    (string, required) The public key of the recipient of the refund.\n"
+            "3. hash          (string, required) SHA256 or RIPEMD160 hash of the preimage.\n"
```

Comment:
Maybe hash should be optional, and if not supplied a `GetRandHash` preimage should be selected and returned to the caller.